ocdata/core/adsorbate_slab_config.py

```diff
@@ -0,0 +1,434 @@
+import copy
+import logging
+from itertools import product
+
+import ase
+import numpy as np
+import scipy
+from ase.data import atomic_numbers, covalent_radii
+from ase.geometry import wrap_positions
+from pymatgen.analysis.adsorption import AdsorbateSiteFinder
+from pymatgen.io.ase import AseAtomsAdaptor
+from scipy.optimize import fsolve
+
+from ocdata.core import Adsorbate, Slab
+from ocdata.core.adsorbate import randomly_rotate_adsorbate
+
+
+class AdsorbateSlabConfig:
+    """
+    Initializes an adsorbate-catalyst system for a given Adsorbate and Slab.
+
+    Arguments
+    ---------
+    slab: Slab
+        Slab object.
+    adsorbate: Adsorbate
+        Adsorbate object.
+    num_sites: int
+        Number of sites to sample.
+    num_augmentations_per_site: int
+        Number of augmentations of the adsorbate per site. Total number of
+        generated structures will be `num_sites` * `num_augmentations_per_site`.
+    adsorbate_height_step_size: float
+        Distance in Angstroms to add to the height of placement incrementally
+        until there is no atomic overlap.
+    """
+
+    def __init__(
+        self,
+        slab: Slab,
+        adsorbate: Adsorbate,
+        num_sites: int = 100,
+        num_augmentations_per_site: int = 1,
+        interstitial_gap: float = 0.1,
+        mode: str = "random",
+    ):
+        assert mode in ["random", "heuristic"]
+
+        self.slab = slab
+        self.adsorbate = adsorbate
+        self.num_sites = num_sites
+        self.num_augmentations_per_site = num_augmentations_per_site
+        self.interstitial_gap = interstitial_gap
+        self.mode = mode
+
+        self.sites = self.get_binding_sites(num_sites)
+        self.atoms_list = self.place_adsorbate_on_sites(
+            self.sites,
+            num_augmentations_per_site,
+            interstitial_gap,
+        )
+        # self.atoms_list = self.filter_unreasonable_structures(self.atoms_list)
```

Comment:
Do we plan on excluding this because Brook's implementation gaurantees(?) we don't run into this scenario